### PR TITLE
✨ Add `SAPBW` source and Prefect tasks

### PR DESCRIPTION
**Pull request by gwieloch**
## Summary
SAP BW connector migration from viadot 1 to viadot 2. Also added some utility functions into viadot utils.

## Importance
Required by the migration project.

## Checklist

This PR:
- [x] follows the guidelines laid out in `CONTRIBUTING.md`
- [ ] links relevant issue(s)
- [x] adds/updates tests (if appropriate)
- [x] adds/updates docstrings (if appropriate)
- [x] adds an entry in `CHANGELOG.md`

### CODE REVIEW

**Comment by trymzet** [comment]
@gwieloch it seems the checks are failing, can you make sure they pass first, then request PR review? Otherwise I would need to check the same code twice.

As for the tests failing due to `pyrfc` not being installed, you need to escape this import in the source, tasks, and tests, same as in the SAP RFC source:
https://github.com/dyvenia/viadot/blob/2.0/src/viadot/sources/sap_rfc.py#L19-L24
https://github.com/dyvenia/viadot/blob/2.0/tests/unit/test_sap_rfc.py#L6-L9
https://github.com/dyvenia/viadot/blob/2.0/src/viadot/orchestration/prefect/tasks/sap_rfc.py#L11-L12

For more info, see the "optional dependencies" sections in the developer guide docs https://viadot.docs.dyvenia.com/developer_guide/

**Review by trymzet — CHANGES_REQUESTED** [review]
Checks not passing

File: tests/integration/test_sap_bw.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    "[Measures].[003YPR44RQTVKWX9T5BFEWUY9]": "order_quantity",
+    "[Measures].[003YPR44RQTVKWX9UW92X7ELV]": "open_orders_quantity",
+    "[Measures].[003YPR44RQTVKWX9YLKYZKH5O]": "number_of_sales_orders",
+    "[Measures].[003YPR44RQTVKWXA51D4J8HRZ]": "number_of_quotations",
+    "[Measures].[003YPR44RQTVKWXAEL7KFSJ6U]": "number_of_orders_created_from_quotations",
+    "[Measures].[003YPR44RQTVKWXARA4PVZ9TK]": "number_of_quotations_expired_validity_date",
+}
+
+
+def test_mindful_to_adls(sap_bw_config_key, adls_credentials_secret):
```
Comment (Rafalz13) [inline tests/integration/test_sap_bw.py:167]:
```suggestion
def test_sap_bw_to_adls(sap_bw_config_key, adls_credentials_secret):
```

File: tests/integration/test_sap_bw.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
+"""'test_sap_bw.py'."""
```
Comment (trymzet) [inline tests/integration/test_sap_bw.py:1]: This should be under orchestration directory.

**Review by trymzet — CHANGES_REQUESTED** [review]
Added my comments. Some unit tests should also be added.

File: src/viadot/sources/sap_bw.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        conn = self._create_connection()
+
+        query = textwrap.wrap(mdx_query, 75)
+        properties = conn.call("RSR_MDX_CREATE_OBJECT", COMMAND_TEXT=query)
+
+        datasetid = properties["DATASETID"]
+        self.query_output = conn.call("RSR_MDX_GET_FLAT_DATA", DATASETID=datasetid)
+        conn.close()
+
+    def _apply_user_mapping(
```
[the remark this answers is not in the document]
Comment (angelika233) [reply src/viadot/sources/sap_bw.py:118]: issue #1066

File: src/viadot/sources/sap_bw.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            mdx_query (str): The MDX query to be passed to connection.
+        """
+        conn = self._create_connection()
+
+        query = textwrap.wrap(mdx_query, 75)
+        properties = conn.call("RSR_MDX_CREATE_OBJECT", COMMAND_TEXT=query)
+
+        datasetid = properties["DATASETID"]
+        self.query_output = conn.call("RSR_MDX_GET_FLAT_DATA", DATASETID=datasetid)
+        conn.close()
```
[the remark this answers is not in the document]
Comment (angelika233) [reply src/viadot/sources/sap_bw.py:116]: issue #1066

File: src/viadot/sources/sap_bw.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+        Args:
+            if_empty (str, optional): What to do if a fetch produce no data.
+                Defaults to "warn
+
+        Raises:
+            ValidationError: Prints the original SAP error message in case of issues
+                with MDX execution.
+            mapping_dict (dict[str, Any], optional): Dictionary with original and new
+                column names. Defaults to None.
```
[the remark this answers is not in the document]
Comment (angelika233) [reply src/viadot/sources/sap_bw.py:161]: #1066

[the remark this answers is not in the document]
Comment (trymzet) [reply src/viadot/sources/sap_bw.py:161]: I mean the docstring, lines 160-161. This shouldn't be under the "Raises" section but under Args.

**Pull request merged**